xmtp_id/src/associations/mod.rs

```diff
@@ -17,26 +17,23 @@ pub use self::signature::{Signature, SignatureError, SignatureKind};
 pub use self::state::{AssociationState, AssociationStateDiff};
 
 // Apply a single IdentityUpdate to an existing AssociationState
-pub fn apply_update(
+pub async fn apply_update(
     initial_state: AssociationState,
     update: IdentityUpdate,
 ) -> Result<AssociationState, AssociationError> {
-    update.update_state(Some(initial_state))
+    update.update_state(Some(initial_state)).await
 }
 
 // Get the current state from an array of `IdentityUpdate`s. Entire operation fails if any operation fails
-pub fn get_state<Updates: AsRef<[IdentityUpdate]>>(
+pub async fn get_state<Updates: AsRef<[IdentityUpdate]>>(
     updates: Updates,
 ) -> Result<AssociationState, AssociationError> {
-    let new_state = updates.as_ref().iter().try_fold(
-        None,
-        |state, update| -> Result<Option<AssociationState>, AssociationError> {
-            let updated_state = update.update_state(state)?;
-            Ok(Some(updated_state))
-        },
-    )?;
-
-    new_state.ok_or(AssociationError::NotCreated)
+    let mut state = None;
+    for update in updates.as_ref().iter() {
+        state = Some(update.update_state(state).await?);
+    }
+
+    state.ok_or(AssociationError::NotCreated)
```

Comment:
I refactored `try_fold` to a for loop, can probably do sth similar to `Future:all`.